event-sourcing/event-publisher/event-publisher-process/src/main/java/uk/gov/justice/services/eventsourcing/prepublish/PrePublishRepository.java

```diff
@@ -16,13 +16,14 @@
 
     private static final int NO_PREVIOUS_EVENT_NUMBER = 0;
     private static final String SELECT_EVENT_NUMBER_SQL = "SELECT event_number FROM event_log WHERE id = ?";
-    private static final String SELECT_PREVIOUS_EVENT_NUMBER_SQL = "SELECT event_number FROM event_log WHERE event_number < ? ORDER BY event_number DESC LIMIT 1";
+    private static final String SELECT_PREVIOUS_EVENT_NUMBER_SQL = "SELECT event_number FROM event_log, event_stream WHERE event_number < ? and event_log.stream_id = event_stream.stream_id and  event_stream.active = true ORDER BY event_number DESC LIMIT 1\n";
     private static final String INSERT_INTO_PUBLISH_QUEUE_SQL = "INSERT INTO publish_queue (event_log_id, date_queued) VALUES (?, ?)";
```

Comment:
Might need to discuss this. I can see why you're doing it, but this query was intended to be as quick as possible; and now it has a join on the event_stream table.

Reply:
Ticket raised to Back log
TP-1160 - To discuss on the impact of getting previous event number